solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/outbound/JCSMPOutboundMessageHandler.java

```diff
@@ -126,7 +168,9 @@ public void handleMessage(Message<?> message) throws MessagingException {
 							targetDestination instanceof Topic ? "TOPIC" : "QUEUE", targetDestination.getName()), e);
 		} finally {
 			if (solaceMeterAccessor != null) {
-				solaceMeterAccessor.recordMessage(properties.getBindingName(), xmlMessage);
+				for (XMLMessage smfMessage : smfMessages) {
+					solaceMeterAccessor.recordMessage(properties.getBindingName(), smfMessage);
```

Comment:
do we have any difference between successfully and not successfully processed messages for the meeter ?